AnkiDroid/src/main/java/com/ichi2/anki/CrashReportService.kt

```diff
@@ -281,12 +282,32 @@ object CrashReportService {
             }
         }
         if (FEEDBACK_REPORT_NEVER != reportMode) {
+            if (!e.safeFromPII()) return
+
             ACRA.errorReporter.putCustomData("origin", origin ?: "")
             ACRA.errorReporter.putCustomData("additionalInfo", additionalInfo ?: "")
             ACRA.errorReporter.handleException(e)
         }
     }
 
+    private fun Throwable.throwableRules() = listOf {
+        // BackendSyncServerMessage may contain PII and we do not want this leaked to ACRA.
+        // Related: https://github.com/ankidroid/Anki-Android/issues/17392
+        // and also https://github.com/ankitects/anki/commit/ba1f5f4
+        if (this is BackendSyncServerMessageException) false else true
+    }
+
+    // Check if the Throwable is safe from Personally Identifiable Information (PII)
+    // based on its class type. Returns false if it isn't, otherwise true.
+    fun Throwable.safeFromPII(): Boolean {
+        val current = throwableRules().all { rule -> rule.invoke() }
+        return if (current && this.cause != null && this.cause != this) {
+            this.cause!!.safeFromPII()
+        } else {
+            current
+        }
+    }
```

Comment:
Optional: the `list` didn't seem to be adding much value with one entry.

This also adds javadocs, which render in the IDE

```suggestion
    /**
     * Checks if the [Throwable] is safe from Personally Identifiable Information (PII)
     * @return `false` if the [Throwable] contains PII, otherwise `true`
     */
    fun Throwable.safeFromPII(): Boolean {
        if (this.containsPIINonRecursive()) return false
        return this.cause?.safeFromPII() != false
    }

    private fun Throwable.containsPIINonRecursive(): Boolean {
        // BackendSyncServerMessage may contain PII and we do not want this leaked to ACRA.
        // Related: https://github.com/ankidroid/Anki-Android/issues/17392
        // and also https://github.com/ankitects/anki/commit/ba1f5f4
        return this is BackendSyncServerMessageException
    }
```